### PR TITLE
itest: logging for subtests and refactor WaitForBlockchainSync

### PR DESCRIPTION
**Pull request by yyforyongyu**
I was reading the test logs to understand why the tests were flaky, and made a few changes in assisting me to locate tests.

### CODE REVIEW

File: lntest/node.go [hunk 1]
```diff
@@ -1079,50 +1079,30 @@ func (hn *HarnessNode) WaitForNetworkChannelClose(ctx context.Context,
...
-			getInfoReq := &lnrpc.GetInfoRequest{}
-			getInfoResp, err := hn.GetInfo(ctx, getInfoReq)
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout while waiting for " +
+				"blockchain sync")
+		case <-hn.quit:
+			return nil
+		case <-ticker.C:
```
Comment (bhandras) [inline lntest/node.go:1097]: This is somewhat different as it won't wait 100 ms between `GetInfo` requests but will wait `100 ms - duration(GetInfo)`.

**Review by bhandras — COMMENTED** [review]
Nice refactors @yyforyongyu :+1:

File: lntest/itest/assertions.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"testing"
+
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/require"
+)
+
+// AddLogAndAssertNoError adds a line to the log file and asserts there's no
+// error.
+func AddLogAndAssertNoError(t *testing.T,
```
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply lntest/itest/assertions.go:14]: Changed. I was also thinking it might be too long😂

File: lntest/node.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/node.go:1097]: Agreed (I think you meant `100ms + duration(GetInfo)`).
Unless we fire the goroutine under `case <-ticker.C`, or in the original code, under `case <-time.After(retryDelay)`, the time to wait between `GetInfo` will always be `100ms + duration(GetInfo)`.

Comment (bhandras) [reply lntest/node.go:1105]: Nice, using the `NewTicker` will also avoid the timer leak that happens when the timer is not fired. One thing where the old more complicated solution was better is that it immediately returned if `hn.quit` was closed regardless if `GetInfo`  was blocking the goroutine but I think overall that is not a big issue and your design is cleaner.

**Review by bhandras — APPROVED** [review]
LGTM :+1:

File: lntest/itest/lnd_test.go [hunk 3]
```diff
@@ -1403,10 +1403,21 @@ test:
 	// new Carol+Dave for each test instance as well.
 	for _, carolCommitType := range allTypes {
 		for _, daveCommitType := range allTypes {
+			carolArgs := carolCommitType.Args()
```
Comment (cfromknecht) [inline lntest/itest/lnd_test.go:1406]: this temp variable (along with `daveArgs`) isn't used in this new code. perhaps they don't need to move?

**Review by cfromknecht — APPROVED** [review]
just one small nit, otherwise looks good to go

File: lntest/itest/lnd_test.go (hunk 3 above)
Comment (yyforyongyu) [reply lntest/itest/lnd_test.go]: yep my bad, now fixed!

**Comment by yyforyongyu** [comment]
> @yyforyongyu can you rebase this on master so we can run the test suite again?

Rebased. Looks like we got a timeout and a flake.

**Comment by Roasbeef** [comment]
Seems to at least not introduce any _new_ issues, so we should land it?

**Comment by Roasbeef** [comment]
Bump.

**Comment by Roasbeef** [comment]
Bump, will move off the 0.13 milestone otherwise.

**Pull request merged**